flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyProtocol.java

```diff
@@ -120,10 +120,14 @@
 	 * @return channel handlers
 	 */
 	public ChannelHandler[] getClientChannelHandlers() {
-		return new ChannelHandler[] {
-			messageEncoder,
-			new NettyMessage.NettyMessageDecoder(),
-			new CreditBasedPartitionRequestClientHandler()};
+			CreditBasedPartitionRequestClientHandler networkClientHandler = new CreditBasedPartitionRequestClientHandler();
+			NettyMessageClientDecoderDelegate nettyMessageClientDecoderDelegate
+					= new NettyMessageClientDecoderDelegate(networkClientHandler);
+
+			return new ChannelHandler[] {
+					messageEncoder,
+				nettyMessageClientDecoderDelegate,
```

Comment:
nit: `new NettyMessageClientDecoderDelegate(networkClientHandler)` instead, avoid above too long definition.

Reply:
Fixed according to the comment.